src/ex2/ItemManager.js

```diff
@@ -0,0 +1,179 @@
+  }
+
+  createPromises(inputArray) {
+    let allPromises = [];
+
+    inputArray.forEach((elm) => {
+      const id = elm.trim();
+
+      allPromises.push(pokemonClient.fetchPokemon(id));
+    });
+
+    return allPromises;
+  }
+
+  removeItem(e, liElm) {
+    e.stopPropagation();
+
+    const itemId = liElm.id.split("-");
+    const pokemonName = itemId[1];
+    const item = itemId.join(" ");
+
+    const itemIndex = this.itemList.findIndex((listItem) => {
+      return listItem === item;
+    });
+
+    this.itemList.splice(itemIndex, 1);
+    this.pokemons.delete(pokemonName);
+
+    this.renderItems();
+  }
+
+  removeAll() {
+    this.itemList.length = 0;
+    this.pokemons.clear();
+
+    this.renderItems();
+  }
+
+  renderItems(current) {
+    this.toggleFooter();
+
+    // clear list innerHTML
+    const list = document.querySelector("#list");
+    list.innerHTML = "";
+
+    // create elements for exisitng items
+    this.itemList.forEach((item) => {
+      const itemNode = this.createItemElement(item, current);
+
+      list.appendChild(itemNode);
+    });
+  }
+
+  toggleFooter() {
+    const addItemButton = document.querySelector("#list-item-submit");
+    const clearAllButton = document.querySelector("#clear-all-button");
+    const sortByNameButton = document.querySelector("#sort-by-name-button");
+
+    if (this.itemList.length === 0) {
+      addItemButton.classList.add("hithere"); //add AddButton animation
+      clearAllButton.classList.add("hidden"); //hide ClearAll button
+      sortByNameButton.classList.add("hidden"); //hide Sort button
+    } else {
+      addItemButton.classList.remove("hithere"); //remove AddButton animation
+      clearAllButton.classList.remove("hidden"); //show ClearAll button
+      sortByNameButton.classList.remove("hidden"); //show Sort button
+    }
+  }
+
+  createItemElement(input, current) {
+    const itemId = input.split(" ").join("-");
+
+    //create list element, add class, innerHTML, eventListener
+    const liElm = document.createElement("li");
+    liElm.setAttribute("id", `${itemId}`);
+    liElm.classList.add("list-item");
+    if (input === current) liElm.classList.add("grow");
+    liElm.innerHTML = input;
+    liElm.addEventListener("click", () => alert(`Task: ${input}`));
+
+    //create delete button, add id, class, src and clickListener
+    const deleteButton = document.createElement("img");
+    deleteButton.setAttribute("id", `${itemId}-delete`);
+    deleteButton.classList.add("list-item-delete-button");
+    deleteButton.src = "../images/delete_icon.svg";
+    deleteButton.addEventListener("click", (e) => this.removeItem(e, liElm));
+    liElm.appendChild(deleteButton);
+
+    return liElm;
+  }
+
+  clearInputField() {
+    document.querySelector("#list-item-input").value = "";
+  }
+
+  sortByName() {
+    this.itemList.sort();
+
+    this.renderItems();
+  }
+
+  capitalize(string) {
+    const updatedString = string.charAt(0).toUpperCase() + string.slice(1);
```

Comment:
better not to concatenate with + operator. 
This is where the `` string interpolation is best used  or even just the concatenate function